source/slang/slang-ir-inline.cpp

```diff
@@ -565,6 +639,58 @@ struct InliningPassBase
             return;
         }
 
+        // If the callee has multiple blocks, use the more complex inlining approach
+        inlineMultipleBlockFuncBody(callSite, env, builder, debugInlinedAt);
+    }
+
+    /// Insert debug scope instructions into all blocks
+    void InsertDebugScopeIntoBlocks(
+        IRFunc* callee,
+        IRBuilder* builder,
+        IRInst* debugInlinedAt)
+    {
+        if (callee->findDecoration<IRDebugLocationDecoration>())
+        {
+            for (auto calleeBlock : callee->getBlocks())
+            {
+                // We keep looping until we can find a place to insert the debug scope.
+                // Note that in the Slang IR, a block may start with IRParams representing 
+                // phi nodes, and we don't want to insert DebugScope before any IRParams.
+                auto inst = calleeBlock->getFirstChild();
+                while (inst)
+                {
+                    if (as<IRDebugLine>(inst) || as<IRParam>(inst))
+                    {
+                        inst = inst->getNextInst();
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+
+                auto inst1 = inst;
+                auto inst2 = inst1->getNextInst();
+                if (inst1 && inst2)
+                {
+                    IRInst* args[] = { debugInlinedAt };
+                    auto instInsert = builder->createIntrinsicInst(builder->getVoidType(), kIROp_DebugScope, 1, args);
```

Comment:
Add a emitDebugScope function to IRBuilder, don’t use createIntrinsicInst directly.